src/charm.py

```diff
@@ -22,14 +27,39 @@
 class KratosCharm(CharmBase):
     """Charmed Ory Kratos."""
 
+    admin_port = 4434
+    public_port = 4433
+
     def __init__(self, *args):
         super().__init__(*args)
         self._container_name = "kratos"
         self._container = self.unit.get_container(self._container_name)
         self._config_file_path = "/etc/config/kratos.yaml"
         self._identity_schema_file_path = "/etc/config/identity.default.schema.json"
 
-        self.service_patcher = KubernetesServicePatch(self, [("admin", 4434), ("public", 4433)])
+        self.service_patcher = KubernetesServicePatch(
+            self, [("admin", self.admin_port), ("public", self.public_port)]
+        )
+        self.admin_ingress = IngressPerAppRequirer(
+            self,
+            relation_name="admin-ingress",
+            port=self.admin_port,
+            strip_prefix=True,
+        )
+        self.public_ingress = IngressPerAppRequirer(
+            self,
+            relation_name="public-ingress",
+            port=self.public_port,
+            strip_prefix=True,
+        )
+
+        # Admin ingress events
```

Comment:
I find it more readable to put all observed events together rather than splitting in ingress, db and pebble-related, but it's just a minor suggestion and you can dismiss it. Let's see what others think